.github/workflows/docker-build-publish.yml

```diff
@@ -0,0 +1,22 @@
+name: Docker Build & Publish
+
+# Trigger on all push events, new semantic version tags, and all PRs
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
+  pull_request:
+
+jobs:
+  docker-security-build:
+    permissions:
+      contents: write
+      packages: write
+    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_pipeline.yml@main # yamllint disable-line rule:line-length
```

Comment:
This will only push on version tags, which we don't have in this repo.

Reply:
@jrmanes - could you please help me out here? also, i'm confused as to why we'd move this to rollkit and then use celestia workflows 🤔

Reply:
hello @jcstein
atm we only have these common pipelines in Celestiaorg, though @smuu is working on it for creating them also in Rollkit. Once we'll have them there, we will have to change the reference to rollkit repo 👌 
hope i've solved you the doubts :)

Reply:
okay, thank you

Reply:
okay. so what do i need to do here? there is a "main" branch on this repo?

Reply:
we will need versioning for this @smuu in the future. 
As all the docs of rollkit mentions how to run tutorials on different celestia testnets. 

Hence why, we also need to replicate some versioning here to not get lost in the weeds.

Reply:
To verify: Does this ticket track your idea @Bidon15?
https://github.com/celestiaorg/devops/issues/174